### PR TITLE
[Docs] [WIP] Add cobra genDocs functionality to autogenerate pages

### PR DESCRIPTION
**Pull request by alphaX86**
Signed-off-by: Aadhitya A <aadhitya864@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**
This PR is WIP and requires lot of reviews. Also, I attached sample files which would show output for docs when the go file is made to run.

P.S: This can tend to fail some CI tests so beware. (Making this PR as draft)

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by warunicorn19** [comment]
Hey @alphaX86 let's have a discussion on what this PR is aiming to do on the Meshery Development call.

**Comment by alphaX86** [comment]
Sure @warunicorn19 here's the slack thread which Lee and I discussed about this (for info) 

https://layer5io.slack.com/archives/CFGG6U10E/p1645479676247319?thread_ts=1645479676.247319&cid=CFGG6U10E

**Comment by warunicorn19** [comment]
Sorry, I missed some comments here, been a bit busy. I'll take a look at the PR by EOD.

**Comment by leecalcote** [comment]
@navendu-pottekkat feedback?

File: mesheryctl/internal/cli/root/root.go [hunk 1]
```diff
@@ -68,6 +68,30 @@ func Execute() {
...
+}
+
+func TreeSingle() *cobra.Command {
+	availableSubcommands = []*cobra.Command{
+		versionCmd,
+	}
+
+	RootCmd.AddCommand(availableSubcommands...)
+	return RootCmd
+}
```
[the remark this answers is not in the document]
Comment (navendu-pottekkat) [reply mesheryctl/internal/cli/root/root.go:93]: @alphaX86 Each of the commands are independent and are in variables like `startCmd`. Could you use that for selecting particular commands?

**Review by navendu-pottekkat — COMMENTED** [review]
@alphaX86 Could we directly generate the doc for the entire tree as mentioned [here](https://github.com/spf13/cobra/blob/master/doc/md_docs.md#generate-markdown-docs-for-the-entire-command-tree)?

Would this docs be generated as a YAML file? Could you run an example and show the output YAML? I'm assuming that this generated YAML will replace the YAMLs you created a while ago and we maintain manually.

Ideally, we should have this running in our CI/CD with changes to mesheryctl updating the docs. Do you think we can achieve that?

**Comment by alphaX86** [comment]
> @alphaX86 Could we directly generate the doc for the entire tree as mentioned [here](https://github.com/spf13/cobra/blob/master/doc/md_docs.md#generate-markdown-docs-for-the-entire-command-tree)?
> 
> Would this docs be generated as a YAML file? Could you run an example and show the output YAML? I'm assuming that this generated YAML will replace the YAMLs you created a while ago and we maintain manually.
> 
> Ideally, we should have this running in our CI/CD with changes to mesheryctl updating the docs. Do you think we can achieve that?

+1 the YAML generated will be indeed different from what I made earlier. The bad part is all will be jumbled according to alphabetical order and not the order we desire. I'll share the output when I'm in my system.

CI can be added but I'm having trouble with golangci-lint in the checks even though I go fmt-ed all files. Any way to leverage that trouble??

File: mesheryctl/internal/cli/root/root.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alphaX86) [reply mesheryctl/internal/cli/root/root.go:93]: Ok I'll try that

**Comment by navendu-pottekkat** [comment]
A more verbose log of the issue will be available if you use golangci-lint on your local machine and you should be able to identify and fix the issue.

**Comment by alphaX86** [comment]
Alright, it said golangci-lint not imported locally for genDocs.go... I'll find a way to fix it.

And yes, we can achieve CI/CD action to autogenerate pages similar to adapter benchmark tests section

**Comment by alphaX86** [comment]
The sample YAML file is generated, do have a look folks... This is generated from the command's data, so if we need any changes, we need to manipulate on the commands directly

**Comment by alphaX86** [comment]
> Let's discuss updates on the call.

I might not be available today due to important work, but if it's resolved, we'll discuss the updates @warunicorn19. I'll let you know before meet

**Comment by alphaX86** [comment]
😫 I forgot this! Will resolve it!

**Comment by warunicorn19** [comment]
@alphaX86 let's discuss today.

**Comment by alphaX86** [comment]
Closing this as this requires extensive work in golang, though the branch in my fork will not be deleted for reference

**Comment by alphaX86** [comment]
#5201

**Pull request closed without merging**